hibernate-core/src/main/java/org/hibernate/type/descriptor/java/OffsetTimeJavaType.java

```diff
@@ -76,31 +76,44 @@ public <X> X unwrap(OffsetTime offsetTime, Class<X> type, WrapperOptions options
 			return null;
 		}
 
+		// for java.time types, we assume that the JDBC timezone, if any, is ignored
+
 		if ( OffsetTime.class.isAssignableFrom( type ) ) {
 			return (X) offsetTime;
 		}
 
+		if ( LocalTime.class.isAssignableFrom( type ) ) {
+			return (X) offsetTime.withOffsetSameInstant( getCurrentSystemOffset() ).toLocalTime();
```

Comment:
Can you add a comment on why you are not using `getCurrentJdbcOffset` here? Same for the `wrap` method.

Reply:
The comment is above:

    // for java.time types, we assume that the JDBC timezone, if any, is ignored

this sort-of doesn't make much sense, unless you know what I have planned:

https://github.com/hibernate/hibernate-orm/pull/5546

Reply:
Aha, yeah that makes sense then, thanks!